homeassistant/helpers/intent.py

```diff
@@ -271,12 +318,21 @@ def __init__(
         self.reprompt: dict[str, dict[str, Any]] = {}
         self.card: dict[str, dict[str, str]] = {}
         self.language = language
+        self.state = IntentResponseState.SUCCESS
+        self.response_type = IntentResponseType.ACTION_CONFIRMATION
+        self.error_reason: IntentResponseErrorReason | None = None
+
+        if self.intent is not None:
+            if self.intent.category == IntentCategory.QUERY:
+                # speech will be the answer to the query
+                self.response_type = IntentResponseType.QUERY_ANSWER
 
     @callback
     def async_set_speech(
         self,
         speech: str,
         speech_type: str = "plain",
+        response_type: IntentResponseType | None = None,
```

Comment:
In what scenario would `set_speech` contain a different response type than that we already derived from the intent ?

I understand that for `reprompt` it might be different.